parallel-install/pkg/config/config.go

```diff
@@ -41,14 +41,22 @@ type Config struct {
 	Version string
 }
 
-// Validate the given configuration options
-func (c *Config) Validate() error {
+// ValidateDeletion verifies that deletion specific options are provided
+func (c *Config) ValidateDeletion() error {
 	if c.WorkersCount <= 0 {
 		return fmt.Errorf("Workers count cannot be <= 0")
 	}
 	if err := c.pathExists(c.ComponentsListFile, "Components list"); err != nil {
 		return err
 	}
+	return nil
+}
+
+// ValidateDeployment verifies that deployment specific options are provided
+func (c *Config) ValidateDeployment() error {
+	if err := c.ValidateDeletion(); err != nil { //deployment requires all core options
```

Comment:
What if someone writes a deletion specific logic in `ValidateDeletion` and forgets that deployment validation also depends on that? Wouldn't it be better to have a base function that both uses for validating the common things?

Reply:
MAybe worth to directly introduce a DeploymentConfig and DeletionConfig object which have decoupled functions

Reply:
Tried to split the configuration into a Deployment- and Deletion specific entity, but the refactoring would become quite huge as the config-object is passed between so many instances. Will track this as improvement for later.

Reply:
The single test function is now split into multiple test functions, one for each public config validation function.